backend/src/controller/defenceController.ts

```diff
@@ -24,46 +24,99 @@ function configValueExceedsCharacterLimit(config: DefenceConfigItem[]) {
 }
 
 function handleDefenceActivation(req: DefenceActivateRequest, res: Response) {
-	const defenceId = req.body.defenceId;
-	const level = req.body.level;
-
-	if (defenceId && level !== undefined) {
-		// activate the defence
-		req.session.levelState[level].defences = activateDefence(
-			defenceId,
-			req.session.levelState[level].defences
-		);
-		res.status(200).send();
-	} else {
-		res.status(400).send();
+	const { defenceId, level } = req.body;
+
+	if (!defenceId) {
+		sendErrorResponse(res, 400, 'Missing defenceId');
+		return;
+	}
+
+	if (!Number.isFinite(level) || level === undefined) {
+		sendErrorResponse(res, 400, 'Missing level');
+		return;
+	}
+
+	if (level < LEVEL_NAMES.LEVEL_1 || level > LEVEL_NAMES.SANDBOX) {
+		sendErrorResponse(res, 400, 'Invalid level');
+		return;
+	}
```

Comment:
We appear to have these same three checks in four different handler functions now. Any way we can extract them?

Reply:
ah yes, I added a method to validate level in my other PR :) just need to switch it here

Reply:
done